apps/rule-manager/app/db/DbRuleDraft.scala

```diff
@@ -350,37 +428,59 @@ object DbRuleDraft extends SQLSyntaxSupport[DbRuleDraft] {
   }
 
   def batchInsert(
-      entities: collection.Seq[DbRuleDraft]
+      entities: collection.Seq[DbRuleDraft],
+      hasNoExternalId: Boolean = false
   )(implicit session: DBSession = autoSession): List[Int] = {
     val params: collection.Seq[Seq[(Symbol, Any)]] = entities.map(entity =>
-      Seq(
-        Symbol("ruleType") -> entity.ruleType,
-        Symbol("pattern") -> entity.pattern,
-        Symbol("replacement") -> entity.replacement,
-        Symbol("category") -> entity.category,
-        Symbol("description") -> entity.description,
-        Symbol("ignore") -> entity.ignore,
-        Symbol("notes") -> entity.notes,
-        Symbol("externalId") -> entity.externalId,
-        Symbol("forceRedRule") -> entity.forceRedRule,
-        Symbol("advisoryRule") -> entity.advisoryRule,
-        Symbol("createdBy") -> entity.createdBy,
-        Symbol("createdAt") -> entity.createdAt,
-        Symbol("updatedBy") -> entity.updatedBy,
-        Symbol("updatedAt") -> entity.updatedAt,
-        Symbol("isArchived") -> entity.isArchived,
-        Symbol("ruleOrder") -> entity.ruleOrder
-      )
+      hasNoExternalId match {
+        case true =>
+          Seq(
```

Comment:
Could we have a common `Seq` here, and conditionally append the extra field, saving us the repetition?

Reply:
I seem to remember I tried to mutate and found that `Seq` has a fixed length, but it looks like I could append to create a new Seq in that case, would be nicer.

Reply:
(Now done)